response.go

```diff
@@ -123,6 +123,7 @@ func (r *Request) Send(c *http.Client) (*Response, error) {
 	if resp.StatusCode >= http.StatusBadRequest {
 		e := &Error{
 			Command: r.Command,
+			Code:    resp.StatusCode,
```

Comment:
This is actually supposed to be the code returned by the commands lib:

https://github.com/ipfs/go-ipfs-cmds/blob/master/error.go#L13-L20

* 0 - normal
* 1 - client error
* 2 - implementation error (e.g., panic)
* 3 - "command not found" (code not used in practice)
* 4 - fatal error (also not used in practice)

It's set in https://github.com/ipfs/go-ipfs-http-client/pull/19/files#diff-b01558cabf60ed66d8a2c08b2bfaba31R138.

---

However, it does make sense to set a default error code for non-json errors.

* http 404 ->  3
* http 400 -> 1
* http 429 -> new error code in the commands lib?

Reply:
I've filed: https://github.com/ipfs/go-ipfs-cmds/pull/168. We'll need a signoff from the JS team.